octoprint_telegram/__init__.py

```diff
@@ -1729,6 +1744,32 @@ def _send_edit_msg(
         except Exception as ex:
             self._logger.debug("Caught an exception in _send_edit_msg(): " + str(ex))
 
+    def take_webcam_snapshot(self, provided_webcam):
+        webcam = provided_webcam.config
+
+        if webcam is None:
+            self._logger.exception("Exception WebcamNotAbleToTakeSnapshotException(provided_webcam is None)")
+            return None
+
+        # using compat.snapshot because snapshotDisplay is supposedly only for user
+        snapshot_url = webcam.compat.snapshot
+        can_snapshot = snapshot_url is not None and snapshot_url != "http://" and snapshot_url != ""
+
+        if not can_snapshot:
+            self._logger.exception("Exception WebcamNotAbleToTakeSnapshotException("+webcam.name+")")
```

Comment:
logging.Logger.exception should only be called from an exception handler: https://docs.python.org/3/library/logging.html#logging.Logger.exception